validate.go

```diff
@@ -208,6 +210,19 @@ func checkLinux(spec rspec.Spec) {
 			logrus.Fatalf("namespace %v is invalid.", spec.Linux.Namespaces[index])
 		} else if spec.Linux.Namespaces[index].Type == rspec.UTSNamespace {
 			utsExists = true
+		} else if spec.Linux.Namespaces[index].Type == rspec.IPCNamespace {
+			ipcExists = true
+		} else if spec.Linux.Namespaces[index].Type == rspec.NetworkNamespace {
+			netExists = true
+		}
+	}
+
+	for k, _ := range spec.Linux.Sysctl {
+		if strings.HasPrefix(k, "net.") && !netExists {
+			logrus.Fatalf("Sysctl %v requires a new Network namespace to be specified as well", k)
+		}
+		if strings.HasPrefix(k, "fs.mqueue.") && !ipcExists {
+			logrus.Fatalf("Sysctl %v requires a new IPC namespace to be specified as well", k)
```

Comment:
Does this also depend on having a `/dev/mqueue` that is specific to your IPC namespace?  That would require a new mount namespace as well (see testing recommendations [here][1]).  And `/dev/mqueue` is not one of the [required filesystems][2], but testing for “will this config supply a new `/dev/mqueue`?” is getting a bit dicey (e.g. maybe the user intends to supply it on their own after `create` and before `start`).  So I think we may want to require a new mount namespace here, but not bother about checking for a new `/dev/mqueue`.

[1]: http://thread.gmane.org/gmane.linux.kernel/782342
[2]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc1/config-linux.md#default-file-systems